charts/redpanda/service_internal.go

```diff
@@ -38,14 +38,26 @@ func ServiceInternal(dot *helmette.Dot) *corev1.Service {
 	// the stateful set and allow the serviceMonitor to target the pods.
 	// This service should not be used by any client application.
 	values := helmette.Unwrap[Values](dot.Values)
-
 	ports := []corev1.ServicePort{}
-	ports = append(ports, corev1.ServicePort{
-		Name:       "admin",
-		Protocol:   "TCP",
-		Port:       values.Listeners.Admin.Port,
-		TargetPort: intstr.FromInt32(values.Listeners.Admin.Port),
-	})
+
+	appProtocol := values.Listeners.Admin.AppProtocol
+
+	if appProtocol != nil {
```

Comment:
There's no need for this check. As `values.Listeners.Admin.AppProtocol` defaults to `nil` and `ServicePort.AppProtocol` also defaults to `nil` you can always include it.

Reply:
this is great, thanks for the feedback, changing now.